lib/ansible/modules/net_tools/cloudflare_dns.py

```diff
@@ -369,7 +369,7 @@ def _cf_simple_api_call(self,api_call,method='GET',payload=None):
 
         if content:
             try:
-                result = json.loads(content)
+                result = json.loads(content.decode('utf-8'))
```

Comment:
We should use to_text here instead of .decode.  to_text is resistant to unicode errors in a way that .decode is not.

``` python

from ansible.module_utils._text import to_native, to_text
[...]
try:
    result = json.loads(to_text(content, errors='surrogate_then_strict'))
except (json.JSONDecodeError, UnicodeError) as e:
    error_msg += "; Failed to parse API response with error {0}: {1}".format(to_native(e), content)